assignments/arrays.js

```diff
@@ -63,31 +63,55 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+ for (let i = 0; i < inventory.length; i++) {
+    if (inventory[i].id == 33) {
+        console.log(`Car 33 is a ${inventory[i].car_year} ${inventory[i].car_make} ${inventory[i].car_model}`);
+    }
+}
+
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+let lastCar = inventory[inventory.length - 1];
+console.log(`Car ${lastCar.id} is a ${lastCar.car_make} ${lastCar.car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+for (let i = 0; i < inventory.length; i++){
+    carModels.push(inventory[i].car_model);
+    carModels.sort();
+}
+   console.log(carModels);
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+for (let i = 0; i < inventory.length; i++){
+    carYears.push(inventory[i].car_year);
+}
+console.log(carYears);
+
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars = [];
-console.log(); 
+for (let i = 0; i < inventory.length; i++){
+    if (inventory[i].car_year < 2000) {
+        oldCars.push(inventory[i].car_year); 
+    }
+}
+console.log(oldCars + oldCars.length); 
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi = [];
+const makes = ['BMW', 'Audi'];
+for (let i = 0; i < inventory.length; i++){
+    if (inventory[i].car_make == 'BMW' || 'Audi') {
+        BMWAndAudi.push(inventory[i]); 
+    }
+}
```

Comment:
I see you made the `makes` array here to hold the values you're checking for, but then used the logical OR operator (`||`) instead.

The way you have it written out right now would cause each car to be pushed into the `BMWAndAudi` array since the if statement is checking whether two statements evaluate to true...

`inventory[i].car_make == 'BMW'`, and `'Audi'`

If you wanted to check equality for both values using the `||` operator, you'd have to write it like this:

```js
for (let i = 0; i < inventory.length; i++){
    if (inventory[i].car_make === 'BMW' || inventory[i].car_make === 'Audi') {
        BMWAndAudi.push(inventory[i]); 
    }
}
```

And of course, you could simplify it further by using the `includes` array method with the makes variable you set:

```js
const makes = ['BMW', 'Audi'];
for (let i = 0; i < inventory.length; i++){
    if (makes.includes(inventory[i].car_make)) {
        BMWAndAudi.push(inventory[i]); 
    }
}
```